client/landing/domains/registrant-verification/index.jsx

```diff
@@ -0,0 +1,249 @@
+			footer: translate( 'All done. You can close this window now.' ),
+			isLoading: false,
+		};
+	};
+
+	getExpiredState = () => {
+		const { translate } = this.props;
+
+		return {
+			message: translate(
+				'This email has expired.{{br/}}Please resend the verification email and try again.',
+				{
+					components: {
+						br: <br />,
+					},
+				}
+			),
+			actionTitle: translate( 'Resend Email' ),
+			actionCallback: this.resendEmail,
+		};
+	};
+
+	getEmailMismatchState = () => {
+		const { domain, email, translate } = this.props;
+
+		return {
+			message: translate(
+				'{{strong}}%(email)s{{/strong}} is different from the email we have on record for {{strong}}%(domain)s{{/strong}}.',
+				{
+					args: {
+						email: email,
+						domain: domain,
+					},
+					components: {
+						strong: <strong />,
+					},
+				}
+			),
+		};
+	};
+
+	getKeySystemsErrorState = errorMessage => {
+		if (
+			'Invalid attribute value; Contact verification already confirmed, nothing to do' ===
+			errorMessage
+		) {
+			const { domain, email, translate } = this.props;
+
+			return {
+				illustration: '/calypso/images/illustrations/illustration-ok.svg',
+				title: translate( 'Already verified.' ),
+				message: translate(
+					"You've already verified {{strong}}%(email)s{{/strong}} for:{{br /}}{{strong}}%(domain)s{{/strong}}.",
+					{
+						args: {
+							email: email,
+							domain: domain,
+						},
+						components: {
+							strong: <strong />,
+							br: <br />,
+						},
+					}
+				),
+				footer: translate( 'All done. You can close this window now.' ),
+			};
+		}
+	};
+
+	getResendEmailErrorState = () => {
+		const { translate } = this.props;
+
+		return {
+			message: translate( "Sorry, we weren't able to re-send the email." ),
+		};
+	};
+
+	getDefaultErrorState = () => {
+		const { translate } = this.props;
+		const defaultErrorFooter = translate(
+			"If you're having trouble verifying your contact information, please {{a}}{{strong}}contact support{{/strong}}{{/a}}.",
+			{
+				components: {
+					a: <a href={ CALYPSO_CONTACT } />,
+					strong: <strong />,
+				},
+			}
+		);
+
+		return {
+			title: translate( 'Uh oh!' ),
+			illustration: '/calypso/images/illustrations/error.svg',
+			message: translate( 'Hmm. Something went wrong.' ),
+			actionTitle: null,
+			actionCallback: null,
+			footer: defaultErrorFooter,
+			isLoading: false,
+		};
+	};
+
+	setErrorState = error => {
+		let errorState;
+
+		switch ( error.error ) {
+			case 'email_mismatch':
+				errorState = this.getEmailMismatchState();
+				break;
+
+			case 'unauthorized':
+				errorState = this.getExpiredState();
+				break;
+
+			case 'Key_Systems_error':
+				errorState = this.getKeySystemsErrorState( error.message );
+				break;
+
+			case 'resend_email_failed':
+				errorState = this.getResendEmailErrorState();
+				break;
+		}
+
+		this.setState( {
+			...this.getDefaultErrorState(),
+			...errorState,
+		} );
+	};
+
+	resendEmail = () => {
```

Comment:
One _potential_ problem with this: for the landing page, we don't require the user to match the blog, etc. (or to be logged in at all). However, `wpcom.resendIcannVerification` _does_ require the user to be logged in and be able to manage the domain.

It might be good to check (in a follow-up PR, this one is big enough) to handle the `unknown_blog` error separately (nudging the user to log in, etc.).

Otherwise, it leads to this slightly misleading and confusing screen (when I tried to resend the verification email for a domain I don't own):
![calypso localhost_3000_domain-services_registrant-verification_domain=transfer-cancel-test-2 com email=5test2@test com token=bf772d0c0cf7902917d367d124c74682 (1)](https://user-images.githubusercontent.com/3392497/55929920-7db7b600-5c0e-11e9-8e1c-352c21b3798a.png)